sourcegraph/jetbrains
# [WIP] Migration to the IntelliJ Platform 2.0.0

Closes https://github.com/sourcegraph/cody-issues/issues/240.

## Test plan

1. Build the plugin (make sure your Java is version 17 or newer):
   ```
   ./gradlew buildPlugin
   ```
2. Open IntelliJ 2024.2 (EAP) and install the plugin manually.
   * The distribution package can be found in `build/distributions`
3. Restart IntelliJ.
4. Verify that the Cody plugin loads without errors, and that the AI chat works.

## Known problems

- Exception at startup: `Cannot create extension (class=com.intellij.ide.misc.PluginAgreementUpdateScheduler)`. The Cody plugin still works though.
- Integration tests become much more flaky, something may be broken
- How do I specify the tool version for the Plugin verifier?

## TODO

- Configure a variant build to support old versions of IntelliJ
